spring-session/src/main/java/org/springframework/session/data/mongo/MongoSessionSerializer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2002-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.mongo;
+
+import com.mongodb.DBObject;
+import org.springframework.session.ExpiringSession;
+
+/**
+ * Interface for serializing and deserializing session objects.
+ * To create custom serializer you have to implement this interface
+ * and simply register your class as a bean.
+ *
+ * @author Jakub Kubrynski
+ */
+public interface MongoSessionSerializer {
```

Comment:
It seems like there should be some existing API (either in Spring Core or Spring Data Mongo) that we could use rather than writing our own. Have you looked into this?